test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -192,14 +228,61 @@ class ExtAuthzGrpcIntegrationTest : public Grpc::GrpcClientIntegrationParamTest,
     EXPECT_EQ("value_1", attributes->destination().labels().at("label_1"));
     EXPECT_EQ("value_2", attributes->destination().labels().at("label_2"));
 
-    // verify headers in check request, making sure that duplicate headers
-    // are merged.
-    EXPECT_EQ("one", (*http_request->mutable_headers())["allowed-prefix-one"]);
-    EXPECT_EQ("two", (*http_request->mutable_headers())["allowed-prefix-two"]);
-    EXPECT_EQ("one,two,three", (*http_request->mutable_headers())["x-duplicate"]);
-    EXPECT_EQ("food", (*http_request->mutable_headers())["regex-food"]);
-    EXPECT_EQ("fool", (*http_request->mutable_headers())["regex-fool"]);
-    EXPECT_FALSE(http_request->headers().contains("not-allowed"));
+    if (encodeRawHeaders()) {
```

Comment:
Add `EXPECT_FALSE(check_request.has_headers());`

And in the else case, add `EXPECT_FALSE(check_request.has_header_map());`